src/evaluators/TryStatement.js

```diff
@@ -23,6 +23,32 @@ import { Value } from "../values/index.js";
 import type { BabelNodeTryStatement } from "babel-types";
 import invariant from "../invariant.js";
 
+// This function gets the evaluated effects with a collection of
+// prior nested affects applied (and their canBeApplied flag reset)
+// We can safely do this as we've wrapped the effects in evaluated
+// effects, meaning all the effects applied to Realm get restored
+
+function evaluateForEffectsWithPriorEffects(
+  realm: Realm,
+  priorEffects: Array<Effects>,
+  f: () => AbruptCompletion | Value,
+  generatorName: string
+): Effects {
+  return realm.evaluateForEffects(
+    () => {
+      for (let priorEffect of priorEffects) realm.applyEffects(priorEffect);
+      let v = f();
```

Comment:
What if `f` throws? Should restoration happen inside a `finally` block, and `f()` be inside of a `try` block?

Reply:
(I'm looking at `withEffectsAppliedInGlobalEnv` which does it this way)